### ISSUE TITLE
Request timed out when calling /Themerr/GetProgress

### ISSUE THREAD

**Issue by haloTT100** [LizardByte/Themerr-jellyfin#332]
### Describe the Bug

I'm getting Unexpected error occurred creating progress dashboard when I navigate to the plugin settings after some time. I think this is caused by because my jellyfin server can't collect all the themes fast enough (I'm using an old laptop for jellyfin). Browser console log shows [Themerr] - undefined
Timeline:
3:35 navigated to the plugin settings, waiting for the dashboard to show up -> Unexpected error occurred creating progress dashboard -> In the jellyfin logs I can see the collected items at: 03:36:10, but it doesn't show up after the error
![error](https://github.com/LizardByte/Themerr-jellyfin/assets/44986488/56aa2cd0-196a-40fb-82ee-90aeba97e4c2)

### Expected Behavior

Show the dashboard and the themes

### Additional Context

OS: Win 7
Jellyfin native Installation: 10.9.1
Plugin Version: 2024.514.151.24
Plugin Installation type: Repository
[jellyfin.log](https://github.com/LizardByte/Themerr-jellyfin/files/15359161/jellyfin.log)
[browser.log](https://github.com/LizardByte/Themerr-jellyfin/files/15359162/browser.log)

**Comment by ReenigneArcher** [comment]
It does look like a timeout issue on the browser's side. I'll look into increasing the default timeout.

```txt
node_modules.jellyfin-apiclient.bundle.js?a32cc7cf54b43478bbe4:2 Request timed out to http://192.168.1.21:8096/jellyfin/Themerr/GetProgress
node_modules.jellyfin-apiclient.bundle.js?a32cc7cf54b43478bbe4:2 Reporting request failure
node_modules.jellyfin-apiclient.bundle.js?a32cc7cf54b43478bbe4:2 Reconnect failed
VM309:253 [Themerr] - Unexpected error occurred creating progress dashboard
VM309:254 [Themerr] - undefined
```

I don't see any error in your server logs, and it seems to generate the list properly.

**Comment by haloTT100** [comment]
I stopped my scheduled tasks letting my processor to take a breath. And it was able to generate the dashboard in time. This does seem like a timeout issue.
![dashboard](https://github.com/LizardByte/Themerr-jellyfin/assets/44986488/d5349a1d-909c-434f-95a5-afeaeee50664)

**Comment by ReenigneArcher** [comment]
Can you try to trigger the timeout again, and let me know how long it takes (in seconds) before it times out?

**Comment by haloTT100** [comment]
Sure, one sec
> It does look like a timeout issue on the browser's side. I'll look into increasing the default timeout.
Wouldn't increasing the default timeout be a temporary fix? I mean with a bigger library the scanning will increasingly be longer, no?

**Comment by haloTT100** [comment]
I timed out after 60 second-ish and it took 85 seconds to collect the whole list

**Comment by ReenigneArcher** [comment]
I just pushed an update to handle this issue. The timeout was not changed, but now it pages through the items. 10 items are requested at a time and there is a loading progress bar so you can see if anything is happening.

**Comment by haloTT100** [comment]
Firstly, thank you for the fast update. Secondly, it seems that I still ocassionally encounter the error when I try to load the dashboard, This probably boils down to my slow latop and I'm not going to stress you to change anything because of my latop's slowness. After some timeout errors it actually shows the dashboard which was my main issue.
Thanks for your work.
![error1](https://github.com/LizardByte/Themerr-jellyfin/assets/44986488/a3a0fda1-33fb-4b85-b5e3-42b9262bbcbd)
![error2](https://github.com/LizardByte/Themerr-jellyfin/assets/44986488/4492538f-c2ee-4c59-908e-b330cc3ab12c)
![done](https://github.com/LizardByte/Themerr-jellyfin/assets/44986488/02633f3c-771f-4116-9dd6-cb7255602e37)

**Comment by ReenigneArcher** [comment]
That's pretty unfortunate. Grabbing only 1o items at a time should really not stress your computer too much.

Do you have a mechanical HDD by chance? Upgrading to an SSD would probably do wonders... as every disk operation would be sped up by 100x or more.

Also, might be worth considering switching to Linux. Windows 7 reached end of life in January 2020. Most Linux distros are known for being less resource intensive than Windows. I'm not one to normally push Linux on anyone, but if your computer is struggling so much I think it might be worth considering.

**Comment by haloTT100** [comment]
I tested it at it seems like if I'm not generating trickplay files it loads the dashboard without issues. Generating those files just eats my CPU and my HDD and it stays at 100%. Sadly, I'm still at 7% generating trickplay for my library so that's going to take some time.
I think my old latop doesn't even support SSD lmao and I don't really want to spend money on such an old computer.
I once tried migrating my jellyfin server to linux, but I'm not really comfortable with that op and I gave up.
Thanks for the advice.
